lib/vagrant/action/builtin/cloud_init_wait.rb

```diff
@@ -0,0 +1,26 @@
+module Vagrant
+  module Action
+    module Builtin
+      class CloudInitWait
+
+        def initialize(app, env)
+          @app    = app
+          @logger = Log4r::Logger.new("vagrant::action::builtin::cloudinitwait")
+        end
+
+        def call(env)
+          machine = env[:machine]
+          if !machine.config.vm.cloud_init_configs.empty?
+            if machine.communicate.test("command -v cloud-init")
+              @logger.info("Waiting for cloud init")
```

Comment:
I think we might want to use the regular UI output for this so users know what the machine is doing.

```suggestion
              env[:ui].output(I18n.t("vagrant.cloud_init_waiting"))
```